module/Finna/src/Finna/RecordDriver/Primo.php

```diff
@@ -272,7 +275,14 @@ public function getLanguages()
      */
     public function getPublicationDates()
     {
-        return $this->fields['date'] ?? [];
+        $result = [];
+        $dates = $this->fields['date'] ?? [];
```

Comment:
empty-tarkistus alla on turha, mutta voisi olla hyvä lisätä (array)-castaus varmuuden vuoksi:
```suggestion
        $dates = (array)($this->fields['date'] ?? []);
```